src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -556,17 +664,74 @@ bool memberHasBadState(Symbol member, LocalState state)
+                                    continue;
+                                }
+
+                                var memberToInitialize = member;
+                                switch (member)
+                                {
+                                    case PropertySymbol:
+                                        // skip any manually implemented properties.
+                                        continue;
+                                    case FieldSymbol { IsConst: true }:
+                                        continue;
+                                    case FieldSymbol { AssociatedSymbol: PropertySymbol prop }:
+                                        // We want to initialize auto-property state to the default state, but not computed properties.
+                                        memberToInitialize = prop;
+                                        break;
+                                    default:
+                                        break;
+                                }
+                                var memberSlot = getSlotForFieldOrPropertyOrEvent(memberToInitialize);
+                                if (memberSlot > 0)
+                                {
+                                    var type = memberToInitialize.GetTypeOrReturnType();
+                                    if (!type.NullableAnnotation.IsOblivious())
+                                    {
+                                        this.State[memberSlot] = type.Type.IsPossiblyNullableReferenceTypeTypeParameter() ? NullableFlowState.MaybeDefault : NullableFlowState.MaybeNull;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        do
+                        {
+                            makeMembersMaybeNull(method, method.NotNullMembers);
+                            makeMembersMaybeNull(method, method.NotNullWhenTrueMembers);
+                            makeMembersMaybeNull(method, method.NotNullWhenFalseMembers);
+                            method = method.OverriddenMethod;
+                        }
+                        while (method != null);
                     }
-                    while (method != null);
+                }
+
+                bool needsDefaultInitialStateForMembers()
+                {
+                    if (_hasInitialState)
+                    {
+                        return false;
+                    }
+
+                    if (!method.HasThisConstructorInitializer() && (!method.ContainingType.IsValueType || method.IsStatic))
+                    {
+                        return true;
+                    }
+
+                    return methodMainNode is BoundConstructorMethodBody ctorBody
+                        && ctorBody.Initializer?.Expression.ExpressionSymbol is MethodSymbol delegatedCtor
```

Comment:
>ctorBody.Initializer?.Expression.ExpressionSymbol is MethodSymbol delegatedCtor [](start = 27, length = 79)

Nit: this could be a pattern in the `is`

Reply:
I did actually do this at one point before committing and I was put off by the amount of curly brace pairs.
`methodMainNode is BoundConstructorMethodBody { Initializer: { Expression: { ExpressionSymbol: MethodSymbol delegatedCtor } } } ctorBody`

It does make me wonder if in a future language version it would make sense to support a "property path" shorthand in recursive patterns, e.g.

`methodMainNode is BoundConstructorBody { Initializer.Expression.ExpressionSymbol: MethodSymbol delegatedCtor }`

Reply:
I'm fairly certain such an issue exists somewhere, but I'm not finding it.

---
In reply to: [470168683](https://github.com/dotnet/roslyn/pull/46441#discussion_r470168683) [](ancestors = 470168683)